packages/juggler/src/loopback-datasource-juggler.ts

```diff
@@ -0,0 +1,662 @@
+
+    delete(options?: Options, callback?: Callback<boolean>): PromiseOrVoid<boolean>;
+
+    /**
+     * Update a single attribute.
+     * Equivalent to `updateAttributes({name: 'value'}, cb)`
+     *
+     * @param {String} name Name of property.
+     * @param {Mixed} value Value of property.
+     * @callback {Function} callback Callback function called with `(err, instance)` arguments.  Required.
+     * @param {Error} err Error object; see [Error object](http://loopback.io/doc/en/lb2/Error-object.html).
+     * @param {Object} instance Updated instance.
+     */
+    updateAttribute(name: string, value: AnyType, options?: Options,
+      callback?: Callback<boolean>): PromiseOrVoid<boolean>;
+
+    /**
+     * Update set of attributes.  Performs validation before updating.
+     *
+     * Triggers: `validation`, `save` and `update` hooks
+     * @param {Object} data Data to update.
+     * @callback {Function} callback Callback function called with `(err, instance)` arguments.  Required.
+     * @param {Error} err Error object; see [Error object](http://loopback.io/doc/en/lb2/Error-object.html).
+     * @param {Object} instance Updated instance.
+     */
+    updateAttributes(data: PersistedData, options?: Options,
+      callback?: Callback<boolean>): PromiseOrVoid<boolean>;
+
+    /**
+     * Replace attributes for a model instance and persist it into the datasource.
+     * Performs validation before replacing.
+     *
+     * @param {Object} data Data to replace.
+     * @options {Object} [options] Options for replace
+     * @property {Boolean} validate Perform validation before saving.  Default is true.
+     * @callback {Function} callback Callback function called with `(err, instance)` arguments.
+     * @param {Error} err Error object; see [Error object](http://loopback.io/doc/en/lb2/Error-object.html).
+     * @param {Object} instance Replaced instance.
+     */
+    replaceAttributes(data: PersistedData, options?: Options,
+      callback?: Callback<boolean>): PromiseOrVoid<boolean>;
+
+    /**
+     * Reload object from persistence.  Requires `id` member of `object` to be able to call `find`.
+     * @callback {Function} callback Callback function called with `(err, instance)` arguments.  Required.
+     * @param {Error} err Error object; see [Error object](http://loopback.io/doc/en/lb2/Error-object.html).
+     * @param {Object} instance Model instance.
+     */
+    reload(options?: Options, callback?: Callback<PersistedData>): PromiseOrVoid<PersistedData>;
+
+    /**
+     * Set the correct `id` property for the `PersistedModel`. Uses the `setId` method if the model is attached to
+     * connector that defines it.  Otherwise, uses the default lookup.
+     * Override this method to handle complex IDs.
+     *
+     * @param {*} val The `id` value. Will be converted to the type that the `id` property specifies.
+     */
+    setId(val: AnyType): void;
```

Comment:
Considering that `EntityRepository ` is already parameterized by `ID` type, can we make `PersistedModel` parameterized by `ID` type too and use `ID` instead of `AnyType` in all methods?